src/mmw/js/src/data_catalog/models.js

```diff
@@ -3,38 +3,122 @@
 var $ = require('jquery'),
     _ = require('lodash'),
     Backbone = require('../../shim/backbone'),
+    moment = require('moment'),
     settings = require('../core/settings');
 
 var REQUEST_TIMED_OUT_CODE = 408;
 var DESCRIPTION_MAX_LENGTH = 100;
 var PAGE_SIZE = settings.get('data_catalog_page_size');
 
-var SearchOption = Backbone.Model.extend({
+
+var FilterModel = Backbone.Model.extend({
     defaults: {
         id: '',
-        active: false,
+        type: '',
+        isValid: true
+    },
+
+    reset: function() {
+        this.clear({ silent: true }).set(this.defaults());
+    },
+
+    validate: function() {
+        _.noop();
+    },
+
+    isActive: function() {
+        window.console.error("Use of unimplemented function",
+                             "FilterModel.isActive");
+        return false;
+    }
+});
+
+var SearchOption = FilterModel.extend({
+    defaults: function() {
+        return _.extend(FilterModel.prototype.defaults, {
+            active: false
+        });
     },
+
+    isActive: function() {
+        return this.get('active');
+    }
 });
 
-var SearchOptions = Backbone.Collection.extend({
-    model: SearchOption,
+var GriddedServicesFilter = SearchOption.extend({
+    defaults: function() {
+        return _.extend(SearchOption.prototype.defaults(), {
+            id: 'gridded',
+            type: 'checkbox',
+            label: 'Gridded Services',
+        });
+    },
+});
+
+var DateFilter = FilterModel.extend({
+    defaults: function() {
+        return _.extend(FilterModel.prototype.defaults, {
+            id: 'date',
+            type: 'date',
+            fromDate: null,
+            toDate: null,
+        });
+    },
+
+    isActive: function() {
+        return this.get('fromDate') || this.get('toDate');
+    },
+
+    validate: function() {
+        // Only need to validate if there are two dates.  Ensure that
+        // before is earlier than after
+        var dateFormat = "MM/DD/YYYY",
+            toDate = this.get('toDate'),
+            fromDate = this.get('fromDate'),
+            isValid = true;
+
+        if (toDate && !moment(toDate, dateFormat).isValid()) {
+            isValid = false;
+        }
+
+        if (fromDate && !moment(fromDate, dateFormat).isValid()) {
+            isValid = false;
+        }
+
+        if (toDate && fromDate){
+            isValid = moment(fromDate, dateFormat)
+                .isBefore(moment(toDate, dateFormat));
+        }
+
+        this.set('isValid', isValid);
+        return isValid;
+    }
+});
+
+var FilterCollection = Backbone.Collection.extend({
+    model: FilterModel,
+
+    countActive: function() {
+        var incrementIfActive = function(count, filter) {
+            return filter.isActive() ? ++count : count;
+        };
+
+        return this.reduce(incrementIfActive, 0);
```

Comment:
This can be made clearer and shorter like this:

```js
var isActive = function(x) { return x.isActive(); };

return this.filter(isActive).length;
```